src/QMCWaveFunctions/TrialWaveFunction.cpp

```diff
@@ -790,8 +793,18 @@ void TrialWaveFunction::flex_updateBuffer(const RefVector<TrialWaveFunction>& wf
   {
     buf_list[iw].get().put(wf_list[iw].get().PhaseValue);
     buf_list[iw].get().put(wf_list[iw].get().LogValue);
-    assert(buf_list[iw].get().size() ==
-           buf_list[iw].get().current() + buf_list[iw].get().current_scalar() * sizeof(double));
+#ifndef NDEBUG
```

Comment:
1. I don't quite get the logic of `#ifndef`.
2. could you write a small helper function so we don't duplicate twice?

Reply:
This shouldn't be checked in production because it should never be wrong. It used to be an assert which only do anything `#ifndef NDEBUG` but it didn't provide the information I needed to debug. I agree about the helper function will do.

Reply:
Thanks. The logic is correct. I was confused by the double negative in the macro condition and name.